WebAPI/ConsoleApp/CategoryRequestGenerator.cs

```diff
@@ -0,0 +1,87 @@
+﻿using MyWebApiApplication.Models;
+using System.Net.Http.Headers;
+using System.Net.Http.Json;
+
+namespace ConsoleApp
+{
+    public class CategoryRequestGenerator
+    {
+        private HttpClient Client { get; set; }
+        public CategoryRequestGenerator()
+        {
+            Client = new HttpClient
+            {
+                BaseAddress = new Uri("https://localhost:7182/api/")
```

Comment:
you can use HTTPCLIENTFACTORY to create httpclient. Follow link: https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests